components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentEvaluatorTest.kt

```diff
@@ -29,68 +29,88 @@ import kotlin.reflect.jvm.isAccessible
+    private lateinit var sharedPrefsOverrideEnabled: SharedPreferences
+    private lateinit var sharedPrefsOverrideBranch: SharedPreferences
+    private lateinit var enabledEditor: SharedPreferences.Editor
+    private lateinit var branchEditor: SharedPreferences.Editor
+    private lateinit var packageManager: PackageManager
+    private lateinit var packageInfo: PackageInfo
+
+    private fun testReset(appId: String = "test.appId", versionName: String = "test.version") {
+        context = mock()
+        sharedPrefsOverrideEnabled = mock()
+        sharedPrefsOverrideBranch = mock()
+        packageManager = mock()
+        packageInfo = PackageInfo()
+
+        `when`(context.packageName).thenReturn(appId)
+
         packageInfo.versionName = "test.version"
         `when`(packageManager.getPackageInfo(anyString(), anyInt())).thenReturn(packageInfo)
         `when`(context.packageManager).thenReturn(packageManager)
 
+        val prefEnabled = ExperimentEvaluator.PREF_NAME_OVERRIDES_ENABLED
+        val prefBranch = ExperimentEvaluator.PREF_NAME_OVERRIDES_BRANCH
+
+        `when`(context.getSharedPreferences(eq(prefEnabled), eq(Context.MODE_PRIVATE))).thenReturn(sharedPrefsOverrideEnabled)
+        `when`(sharedPrefsOverrideEnabled.getBoolean(anyString(), anyBoolean())).thenAnswer { invocation -> invocation.arguments[1] as Boolean }
+        enabledEditor = mock()
+        `when`(enabledEditor.putBoolean(anyString(), anyBoolean())).thenReturn(enabledEditor)
+        `when`(enabledEditor.remove(anyString())).thenReturn(enabledEditor)
+        `when`(enabledEditor.clear()).thenReturn(enabledEditor)
+        `when`(sharedPrefsOverrideEnabled.edit()).thenReturn(enabledEditor)
+
+        `when`(context.getSharedPreferences(eq(prefBranch), eq(Context.MODE_PRIVATE))).thenReturn(sharedPrefsOverrideBranch)
+        `when`(sharedPrefsOverrideBranch.getString(anyString(), anyString())).thenAnswer { invocation -> invocation.arguments[1] as String }
+        branchEditor = mock()
+        `when`(branchEditor.putString(anyString(), anyString())).thenReturn(branchEditor)
+        `when`(branchEditor.remove(anyString())).thenReturn(branchEditor)
+        `when`(branchEditor.clear()).thenReturn(branchEditor)
+        `when`(sharedPrefsOverrideBranch.edit()).thenReturn(branchEditor)
+    }
+
+    private fun setOverride(experimentName: String, isActive: Boolean, branchName: String) {
+        `when`(sharedPrefsOverrideEnabled.getBoolean(eq(experimentName), anyBoolean())).thenReturn(isActive)
+        `when`(sharedPrefsOverrideBranch.getString(eq(experimentName), eq(null))).thenReturn(branchName)
+        `when`(sharedPrefsOverrideEnabled.contains(eq(experimentName))).thenReturn(true)
+        `when`(sharedPrefsOverrideBranch.contains(eq(experimentName))).thenReturn(true)
+    }
+
+    @Test
+    fun `evaluate empty matchers`() {
+        val experiment = testDefaultExperiment(
+            id = "testexperiment",
+            lastModified = 1528916183,
```

Comment:
Could you comment as to what these magic numbers are so that someone looking at this test has a little better context as to what is being tested and how?